### DISCUSSION TITLE
sea-orm-cli generates incorrect model for Postgres database containing 'tsvector' columns

### DISCUSSION THREAD

**Discussion by flyaruu** [SeaQL/sea-orm#1880] [General]
## Description

I'm using a **postgres** data model I tend to use for demos (some fictional data about movie rentals), and one of the tables has a column of type 'tsvector' 

When I generate the ORM source, it looks like this:
```
#[sea_orm(column_type = "custom(\"tsvector\")")]
pub fulltext: String,
```

(The data + data model is open source, available [here](https://github.com/flyaruu/flink-movie-demo/blob/74ce3bc7634fec6359976a4f727c429ed93fa574/film/scripts/d_film.sql#L112): but I don't think it is anything special.

When querying a record of this type I get this panic:
```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Query(SqlxError(ColumnDecode { index: "\"fulltext\"", source: "mismatched types; Rust type `core::option::Option<alloc::string::String>` (as SQL type `TEXT`) is not compatible with SQL type `tsvector`" }))',
```
So it looks like String isn't the right type, but I'm unsure what is, and if sqlx supports it at all. I can comment out this column for now, then it all works. 

But I'd say it would be better to ignore this type, as it breaks the entire table.

## Steps to Reproduce

1. Take a database containing a tsvector column (e.g. the docker image I reference)
2. Generate a model using sea-orm-cli 
3. Query one record (I did a find_by_id, but I don't think it matters)
4. Panic!

### Expected Behavior

The result of that type

### Actual Behavior

A panic

### Reproduces How Often

Always

## Versions
```
├── sea-orm v0.11.0
│   ├── sea-orm-macros v0.11.0 (proc-macro)
│   ├── sea-query v0.28.3
│   │   ├── sea-query-derive v0.3.0 (proc-macro)
│   ├── sea-query-binder v0.3.0
│   │   ├── sea-query v0.28.3 (*)
│   ├── sea-strum v0.23.0
│   │   └── sea-strum_macros v0.23.0 (proc-macro)
```

**Comment by tyt2y3** [comment]
I never used `tsvector`, but if you know what type it is compatible with, you can try the `#[sea_orm(select_as)]` macro attribute.
If you want to ignore it, just add `#[sea_orm(ignore)]`

**Comment by cbio-thomas** [comment]
Not an answer, but work around when dealing with a table with `tsvector` type:
```rs
        #[derive(Debug, FromQueryResult)]
        struct MovieSearchResult {
            id: i64,
            name: String,
            desc: String,
        }

        let res = MovieSearchResult::find_by_statement(Statement::from_sql_and_values(
            DbBackend::Postgres,
            r#"
                select id, name, desc
                from cinema.movies
                where desc_tsvector @@ to_tsquery('english', $1);
            "#,
            [search_str.into()],
        ))
        .all(&conn)
        .await;
```

**Comment by anshap1719** [comment]
Not sure if anyone is still interested, but I believe this first needs to be supported by `sqlx` and you can track the status for that here: https://github.com/launchbadge/sqlx/issues/2705

Also, we need support from sqlx because we also need to create queries in `tsquery` format, which will be the next step.

**Reply by zhabiba24** [reply]
This would be useful once sqlx support arrives

**Reply by anshap1719** [reply]
@zhabiba24 I've since abandoned that work on the SQLx side unfortunately. 😅

**Comment by Huliiiiii** [comment]
Closed in favour of #2750
